api/v1/syncedsecret_types.go

```diff
@@ -104,6 +105,8 @@ type SyncedSecret struct {
 
 	Spec   SyncedSecretSpec   `json:"spec,omitempty"`
 	Status SyncedSecretStatus `json:"status,omitempty"`
+
+	Type corev1.SecretType `json:"type,omitempty"`
```

Comment:
This property should probably live within `struct SyncedSecretSpec` since it defines the generated `Secret`, any strong reason to have this within `SyncedSecret` ?

Reply:
Mainly limited understanding of the operatoe pattern. I will move it up into the spec.